### DISCUSSION TITLE
Generating a side navbar from markdown headers

### DISCUSSION THREAD

**Discussion by OfficePop** [gatsbyjs/gatsby#31911] [Help]
I'm using the `gatsby-transformer-remark` plugin to generate content in my product's site from markdown files, along with `gatsby-remark-autolink-headers` to add anchor links for each header in an article. This has been great, but it would really helpful to add a side navbar to our articles that pulls in the anchor links from the headers and displays them like a TOC the user can refer to jump to various parts of the article as they see fit. 

Given what i'm already doing I figure this must be possible, but i'm not an enough an expert in gatsby or react to do this on my own. Does anyone know if there's a plug-in that can do what i'm asking for, or a guide that can help me designing a component for this?

**Comment by cangSDARM** [comment]
see another discussion: https://github.com/gatsbyjs/gatsby/discussions/32171
